src/main/scala/io/shiftleft/fuzzyc2cpg/AstVisitor.scala

```diff
@@ -42,7 +42,12 @@ class AstVisitor(outputModuleFactory: CpgOutputModuleFactory, structureCpg: CpgS
                                                   graphAdapter)
     astToCfgConverter.convert(functionDef)
 
-    outputModule.persistCpg(bodyCpg)
+    if (functionDef.isOnlyDeclaration) {
+      FuzzyC2CpgCache.add(functionDef.getFunctionSignature, outputIdentifier, bodyCpg)
```

Comment:
Do we need to persist the CPG in this branch?

Reply:
No. In this branch, we only store the declaration. Once we're done with concurrent processing, we then iterate over stored declarations and persist them. Let me add a comment though.

Reply:
Added a comment